panic: concurrent write to websocket connection

#### Version information:
```
go-ipfs version: 0.4.22-
Repo version: 7
System version: amd64/linux
Golang version: go1.12.7
```
#### Description:
I was running the daemon in the foreground, and after a while this happened. I might have been using the WebUI.

```
➜  gen git:(master) ✗ ipfs daemon
Initializing daemon...
go-ipfs version: 0.4.22-
Repo version: 7
System version: amd64/linux
Golang version: go1.12.7
Swarm listening on /ip4/10.14.10.6/tcp/4001
Swarm listening on /ip4/127.0.0.1/tcp/4001
Swarm listening on /ip4/172.17.0.1/tcp/4001
Swarm listening on /ip4/172.18.0.1/tcp/4001
Swarm listening on /ip4/172.19.0.1/tcp/4001
Swarm listening on /ip4/192.168.0.3/tcp/4001
Swarm listening on /ip6/2a02:8109:98c0:1aa5:5c8b:382:9bfb:b75f/tcp/4001
Swarm listening on /ip6/2a02:8109:98c0:1aa5:5e60:26f9:93b:82fd/tcp/4001
Swarm listening on /ip6/::1/tcp/4001
Swarm listening on /p2p-circuit
Swarm announcing /ip4/10.14.10.6/tcp/4001
Swarm announcing /ip4/127.0.0.1/tcp/4001
Swarm announcing /ip4/172.17.0.1/tcp/4001
Swarm announcing /ip4/172.18.0.1/tcp/4001
Swarm announcing /ip4/172.19.0.1/tcp/4001
Swarm announcing /ip4/192.168.0.3/tcp/4001
Swarm announcing /ip6/2a02:8109:98c0:1aa5:5c8b:382:9bfb:b75f/tcp/4001
Swarm announcing /ip6/2a02:8109:98c0:1aa5:5e60:26f9:93b:82fd/tcp/4001
Swarm announcing /ip6/::1/tcp/4001
API server listening on /ip4/127.0.0.1/tcp/5001
WebUI: http://127.0.0.1:5001/webui
Gateway (readonly) server listening on /ip4/127.0.0.1/tcp/8080
Daemon is ready
panic: concurrent write to websocket connection

goroutine 954106 [running]:
github.com/gorilla/websocket.(*messageWriter).flushFrame(0xc0011879b0, 0x1, 0x0, 0x0, 0x0, 0xc004a08966, 0xc007b7f4f0)
        pkg/mod/github.com/gorilla/websocket@v1.4.0/conn.go:591 +0x77a
github.com/gorilla/websocket.(*messageWriter).Close(0xc0011879b0, 0x97bc38, 0x10)
        pkg/mod/github.com/gorilla/websocket@v1.4.0/conn.go:709 +0x56
github.com/gorilla/websocket.(*Conn).prepWrite(0xc00231f600, 0x2, 0x0, 0xc007b7f500)
        pkg/mod/github.com/gorilla/websocket@v1.4.0/conn.go:459 +0x20d
github.com/gorilla/websocket.(*Conn).NextWriter(0xc00231f600, 0x2, 0x7fdcd8d47100, 0xc001c8cc80, 0xc001c8cc80, 0x1007fdcd8d47100)
        pkg/mod/github.com/gorilla/websocket@v1.4.0/conn.go:494 +0x39
github.com/gorilla/websocket.(*Conn).WriteMessage(0xc00231f600, 0x2, 0xc001f5be40, 0x22, 0x40, 0x10, 0xc004714cc0)
        pkg/mod/github.com/gorilla/websocket@v1.4.0/conn.go:755 +0x73
github.com/libp2p/go-ws-transport.(*Conn).Write(0xc0011865d0, 0xc001f5be40, 0x22, 0x40, 0x163e9e0, 0x17968e0, 0x7fdcd8d25800)
        pkg/mod/github.com/libp2p/go-ws-transport@v0.0.4/conn.go:74 +0x59
bytes.(*Buffer).WriteTo(0xc001128fc0, 0x7fdcda72b0b0, 0xc0011877a0, 0x7fdcd8d25800, 0xc001128fc0, 0xe26301)
        /usr/lib/go/src/bytes/buffer.go:242 +0xb8
io.copyBuffer(0x7fdcda72b0b0, 0xc0011877a0, 0x1a99760, 0xc001128fc0, 0x0, 0x0, 0x0, 0x1947720, 0x2, 0x2)
        /usr/lib/go/src/io/io.go:384 +0x34e
io.Copy(...)
        /usr/lib/go/src/io/io.go:364
github.com/multiformats/go-multistream.SelectProtoOrFail.func1(0x17c7a9d, 0xc, 0x7fdcda72aff8, 0xc0011877a0, 0xc00132dec0)
        pkg/mod/github.com/multiformats/go-multistream@v0.0.4/client.go:27 +0x164
created by github.com/multiformats/go-multistream.SelectProtoOrFail
        pkg/mod/github.com/multiformats/go-multistream@v0.0.4/client.go:23 +0x95
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -6,13 +6,13 @@ aliases:
   restore_gomod: &restore_gomod
     restore_cache:
       keys:
-        - v4-dep-{{ .Branch }}-{{ checksum "~/ipfs/go-ipfs/go.sum" }}-{{ .Environment.CIRCLE_JOB }}
-        - v4-dep-{{ .Branch }}-{{ checksum "~/ipfs/go-ipfs/go.sum" }}-
-        - v4-dep-{{ .Branch }}-
-        - v4-dep-master-
+        - v5-dep-{{ .Branch }}-{{ checksum "~/ipfs/go-ipfs/go.sum" }}-{{ .Environment.CIRCLE_JOB }}
+        - v5-dep-{{ .Branch }}-{{ checksum "~/ipfs/go-ipfs/go.sum" }}-
+        - v5-dep-{{ .Branch }}-
+        - v5-dep-master-
   store_gomod: &store_gomod
       save_cache:
-        key: v4-dep-{{ .Branch }}-{{ checksum "~/ipfs/go-ipfs/go.sum" }}-{{ .Environment.CIRCLE_JOB }}
+        key: v5-dep-{{ .Branch }}-{{ checksum "~/ipfs/go-ipfs/go.sum" }}-{{ .Environment.CIRCLE_JOB }}
         paths:
           - ~/go/pkg/mod
           - ~/.cache/go-build/

--- a/go.mod
+++ b/go.mod
@@ -90,6 +90,7 @@ require (
 	github.com/multiformats/go-multiaddr-net v0.1.1
 	github.com/multiformats/go-multibase v0.0.1
 	github.com/multiformats/go-multihash v0.0.10
+	github.com/multiformats/go-multistream v0.1.1 // indirect
 	github.com/opentracing/opentracing-go v1.1.0
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.4.0

